sympy/combinatorics/fp_groups.py

```diff
@@ -69,13 +69,59 @@ def __new__(cls, fr_grp, relators):
         obj._is_standardized = False
 
         obj._order = None
+        obj._parent = None
         obj._center = None
         return obj
 
     @property
     def free_group(self):
         return self._free_group
 
+    @property
+    def parent(self):
+        return self._parent
+
+    @property
+    def relators(self):
+        return tuple(self._relators)
+
+    def _generators(self):
+        return self.free_group.generators
+
+    def subgroup(self, gens):
+        '''
+        Return the subgroup generated by `gens` using the
+        Reidemeister-Schreier algorithm
+
+        '''
+        if not all([isinstance(g, FreeGroupElement) for g in gens]):
+            raise ValueError("Generators must be `FreeGroupElement`s")
+        letters = self.free_group.symbols
+        for g in gens:
+            c = [t[0] in letters for t in g.array_form]
+            if not all(c):
```

Comment:
Each FreeGroupElement has attribute `group`. Perhaps we should use the condition `g.group == self.free_group`.